src/components/workflow/workflow.service.ts

```diff
@@ -0,0 +1,944 @@
+              value: session.token
+            }),
+            relation('in', '', 'token', {
+              active: true,
+            }),
+            node('requestingUser', 'User', {
+              id: session.userId
+            }),
+            relation('in', '', 'member'),
+            node('sg', 'SecurityGroup', {
+              id: input.securityGroupId
+            }),
+          ],
+          [
+            node('requestingUser'),
+            relation('in', '', 'admin', {
+              active: true
+            }),
+            node('baseNode', 'BaseNode')
+          ],
+          [
+            node('state', 'State', {
+              id: input.stateId
+            })
+          ]
+        ])
+        .merge([
+          [
+            node('state'),
+            relation('out', '', 'notification', {
+              active: true,
+              onEnter: true
+            }),
+            node('sg')
+          ]
+        ])
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not attach security group to state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // removeNotificationGroupFromState
+  async removeNotificationGroup(session: ISession, input: GroupState): Promise<void>{
+    try{
+      await this.db
+        .query()
+        .match([
+          [
+            node('token', 'Token', {
+              active: true,
+              value: session.token
+            }),
+            relation('in', '', 'token', {
+              active: true,
+            }),
+            node('requestingUser', 'User', {
+              id: session.userId
+            }),
+            relation('in', '', 'member'),
+            node('sg', 'SecurityGroup', {
+              id: input.securityGroupId
+            })
+          ],
+          [
+            node('requestingUser'),
+            relation('in', '', 'admin', {
+              active: true
+            }),
+            node('baseNode', 'BaseNode')
+          ],
+          [
+            node('state', 'State', {
+              id: input.stateId
+            }),
+            relation('out', 'rel', 'notification'),
+            node('sg')
+          ]
+        ])
+        .detachDelete('rel')
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not remove security group from state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // changeCurrentStateInWorkflow
+  async changeCurrentState(session: ISession, input: ChangeState): Promise<CommentState>{
+    try{
+      const state = await this.readOneState(session, input.newStateId) as State;
+      await this.db
+        .query()
+        .match([
+          [
+            ...matchSession(session),
+            relation('in', '', 'member', {
+              admin: true
+            }),
+            node('sg', 'SecurityGroup'),
+            relation('out', '', 'permission', {
+              write: true
+            }),
+            node('permission', 'Permission'),
```

Comment:
You need to specify that the permission node has a value of the property you are asking about and that the canWrite property is true. 

Something like:

`node('permission', 'Permission', {value: currentState, canWrite: true})`

Otherwise, you aren't checking if the user has permission to write. Look at the schema for Security Groups and Permissions.